src/plugins/discover/public/application/main/utils/update_search_source.ts

```diff
@@ -35,20 +35,27 @@ export function updateVolatileSearchSource(
     uiSettings.get(SORT_DEFAULT_ORDER_SETTING)
   );
   const useNewFieldsApi = !uiSettings.get(SEARCH_FIELDS_FROM_SOURCE);
-  searchSource.setField('trackTotalHits', true).setField('sort', usedSort);
+  const { filterManager, timefilter } = data.query;
+  const filters = [...filterManager.getGlobalFilters()];
```

Comment:
Looks like now pinned filters are being added twice:

<img width="500" alt="Screenshot 2023-06-28 at 09 19 33" src="https://github.com/elastic/kibana/assets/1415710/67062aae-c6e6-4ad7-bc2a-3d081d043069">

This PR change helps to fix requests on page load but creates duplicates in requests when filters get changed to pinned. 

I tested the main and noticed that pinned filters are actually included in requests. It works while the user is changing filters. Pinned filters are missing only after reloading the page.

I think it needs more digging since it's partially working on main. The issue might be in a different place.

Reply:
>This PR change helps to fix requests on page load but creates duplicates in requests when filters get changed to pinned.

Good catch, I'm surprised this happens and thought the search layer would dedupe them before sending the request.

>I tested the main and noticed that pinned filters are actually included in requests. It works while the user is changing filters. Pinned filters are missing only after reloading the page.

Yes, I was of aware of this already and made sure to include this scenario in the new functional tests I added. I don't think it's only on reload, but reloading is the easiest way to reproduce it.

>I think it needs more digging since it's partially working on main. The issue might be in a different place.

This was my concern with splitting the other PR apart since this fixed was designed to work specifically in combination with the other change. I _really_ think the correct solution to this issue is to not include pinned filters in saved searches at all, but I can explore a different fix instead.

Reply:
Alright, let's try this again from the start 😅 I implemented an entirely new fix completely separate from the the other PR that should fix the issue consistently without duplicating filters.